lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/task/CommunicationGroupClientImpl.java

```diff
@@ -191,6 +248,8 @@ public void initialize() {
         throw new RuntimeException("InterruptedException while waiting for initialization", e);
       }
 
+      updateTopology();
```

Comment:
You're right. I included this statement to collect the list of active tasks during initialization, but forgot  the fact that only the `ScatterSender` needs to do this. I'll refactor this class so that only `ScatterSenders` perform this update during initialization.